### PR TITLE
Added support for list inputs

### PR DESCRIPTION
**Pull request by gshashank84**
```y_pred.shape.ndims``` -> Not supports **list** as y_pred variable
 Whereas ```array_ops.rank(y_pred).numpy()```  works for all (Tensors, Numpy arrays, lists)

From #27150

### CODE REVIEW

File: tensorflow/python/keras/metrics.py [hunk 1]
```diff
@@ -2273,10 +2273,10 @@ def update_state(self, y_true, y_pred, sample_weight=None):
       Update op.
     """
     # Flatten the input if its rank > 1.
-    if y_pred.shape.ndims > 1:
+    if array_ops.rank(y_pred).numpy() > 1:
```
Comment (pavithrasv) [inline tensorflow/python/keras/metrics.py]: Can we convert to tensor instead?

Comment (gshashank84) [reply tensorflow/python/keras/metrics.py]: yes, at first I thought of that only.

**Review by gshashank84 — COMMENTED** [review]
I have changed and now they are converted into tensors. 
Have a look @pavithrasv

File: tensorflow/python/keras/metrics.py [hunk 2]
```diff
@@ -2272,6 +2272,10 @@ def update_state(self, y_true, y_pred, sample_weight=None):
     Returns:
       Update op.
     """
+    # Converted y_true and y_pred into Tensors
+    y_true = ops.convert_to_tensor(y_true)
+    y_pred = ops.convert_to_tensor(y_pred)
+
```
Comment (pavithrasv) [inline tensorflow/python/keras/metrics.py]: Can we add a test case for list inputs? Sorry, i forgot to mention this in my first review.

Comment (gshashank84) [reply tensorflow/python/keras/metrics.py]: Sure. You mean checking for list inputs, right?

File: tensorflow/python/keras/metrics.py (hunk 1 above)
Comment (pavithrasv) [reply tensorflow/python/keras/metrics.py]: Yes, a test case for the use case that wasn't supported before this change.

**Review by gshashank84 — COMMENTED** [review]
I think I have added the test case, any other thing that needed to be added? @pavithrasv 

Waiting eagerly for my first PR to merge in TF :smile:

File: tensorflow/python/keras/metrics.py [hunk 3]
```diff
@@ -2272,9 +2272,14 @@ def update_state(self, y_true, y_pred, sample_weight=None):
...
-    y_true = ops.convert_to_tensor(y_true)
-    y_pred = ops.convert_to_tensor(y_pred)
+    
+    # If y_true and y_pred are lists then 
+    # they are converted into Tensors 
+    if type(y_true) == list:
+      y_true = ops.convert_to_tensor(y_true)
+      
+    if type(y_pred) == list:
+      y_pred = ops.convert_to_tensor(y_pred)
```
Comment (gshashank84) [inline tensorflow/python/keras/metrics.py]: I am sorry to interrupt, but do they look good? @pavithrasv

Comment (pavithrasv) [reply tensorflow/python/keras/metrics.py]: Thank you @gshashank84. May be what we need is: 
    y_true = math_ops.cast(y_true, self._dtype)
    y_pred = math_ops.cast(y_pred, self._dtype)
Please take a look at the other metric classes `update_state` function.

**Review by gshashank84 — COMMENTED** [review]
Thanks @pavithrasv for suggesting the changes need to make, I have made a commit now. Would you mind reviewing?

**Comment by pavithrasv** [comment]
> I think I have added the test case, any other thing that needed to be added? @pavithrasv
> 
> Waiting eagerly for my first PR to merge in TF

I am not seeing the test file in the PR. Did you forget to add that?

**Comment by gshashank84** [comment]
> > I think I have added the test case, any other thing that needed to be added? @pavithrasv
> > Waiting eagerly for my first PR to merge in TF
> 
> I am not seeing the test file in the PR. Did you forget to add that?

yes, I forget to add the test file. Do I need to make a new PR with all these included in it?

**Comment by pavithrasv** [comment]
> > > I think I have added the test case, any other thing that needed to be added? @pavithrasv
> > > Waiting eagerly for my first PR to merge in TF
> > 
> > 
> > I am not seeing the test file in the PR. Did you forget to add that?
> 
> yes, I forget to add the test file. Do I need to make a new PR with all these included in it?

No, you can just add it here.

**Comment by gshashank84** [comment]
Thanks for the help @pavithrasv , I have addded the test file and made the changes. :smile: 
Previously in the test file the list was converted into the tensor and then passed into the update_state method but now they are passed straight away to test the update_state of MeanIoU.

File: tensorflow/python/keras/metrics.py [hunk 4]
```diff
@@ -2272,6 +2272,12 @@ def update_state(self, y_true, y_pred, sample_weight=None):
     Returns:
       Update op.
     """
+    
+    # If y_true and y_pred are lists then
```
Comment (pavithrasv) [inline tensorflow/python/keras/metrics.py]: nit: I think we can remove these lines.

**Review by pavithrasv — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/metrics.py (hunk 4 above)
Comment (gshashank84) [reply tensorflow/python/keras/metrics.py]: removed!

**Comment by gshashank84** [comment]
ready to reviewed:) @pavithrasv

**Review by pavithrasv — APPROVED** [review]
